validator/testcases/javascript/actions.py

```diff
@@ -643,33 +651,36 @@ def _expr_assignment(traverser, node):
                     # If it's a global and the actual member exists, test
                     # whether it can be safely overwritten.
                     member = member_object_value["value"][member_property]
-                    readonly_value = (member["readonly"] if
-                                      "readonly" in member else
-                                      True)
+                    if "readonly" in member:
+                        global_overwrite = True
+                        readonly_value = member["readonly"]
 
         traverser._debug("ASSIGNMENT:DIRECT:GLOB_OVERWRITE %s" %
-                             global_overwrite)
+                         global_overwrite)
+        traverser._debug("ASSIGNMENT:DIRECT:READONLY %r" %
+                         readonly_value)
 
-        if (global_overwrite and
-            not traverser.is_jsm and
-            not traverser.err.get_resource("em:bootstrap") and
-            readonly_value == True):
+        if callable(readonly_value):
+            # The readonly attribute supports a lambda function that accepts
+            readonly_value = readonly_value(t=traverser, r=right,
+                                            rn=node["right"])
 
-            traverser.err.warning(
+        if readonly_value and global_overwrite:
+
+            kwargs = dict(
                 err_id=("testcases_javascript_actions",
                         "_expr_assignment",
                         "global_overwrite"),
                 warning="Global variable overwrite",
                 description="An attempt was made to overwrite a global "
-                            "variable in some JavaScript code.",
-                filename=traverser.filename,
-                line=traverser.line,
-                column=traverser.position,
-                context=traverser.context)
```

Comment:
On Tue, May 05, 2015 at 08:00:20AM -0700, Mathieu Agopian wrote:
>>                  err_id=("testcases_javascript_actions",
>>                          "_expr_assignment",
>>                          "global_overwrite"),
>>                  warning="Global variable overwrite",
>>                  description="An attempt was made to overwrite a global "
>> -                            "variable in some JavaScript code.",
>> -                filename=traverser.filename,
>> -                line=traverser.line,
>> -                column=traverser.position,
>> -                context=traverser.context)
>
>We don't need those anymore?

No. mstriemer added these utility functions to `Traverser` a 
while ago to take care of those parameters, and has been slowly 
fixing legacy calls since.